src/python/pants/backend/experimental/visibility/register.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.internals.dep_rules import (
+    BuildFileDependencyRules,
+    BuildFileDependencyRulesImplementation,
+    BuildFileDependencyRulesImplementationRequest,
+)
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import (
+    Dependencies,
+    DependenciesRuleAction,
+    DependenciesRuleActionRequest,
+    FieldSet,
+    ValidatedDependencies,
+    ValidateDependenciesRequest,
+)
+from pants.engine.unions import UnionRule
+
+
+class BuildFileVisibilityImplementationRequest(BuildFileDependencyRulesImplementationRequest):
+    pass
+
+
+class VisibilityValidateFieldSet(FieldSet):
+    required_fields = (Dependencies,)
+
+
+class VisibilityValidateDependenciesRequest(ValidateDependenciesRequest):
+    field_set_type = VisibilityValidateFieldSet
+
+
+@rule
+def build_file_visibility_implementation(
+    _: BuildFileVisibilityImplementationRequest,
+) -> BuildFileDependencyRulesImplementation:
+    return BuildFileDependencyRulesImplementation(BuildFileDependencyRules)
```

Comment:
I'm a little confused - I don't see anything visibility-specific here?

Reply:
True, that's because the base class implements it, which is a sign that it probably should have no implementation, and move that here:

https://github.com/pantsbuild/pants/pull/17401/files#diff-3e00979219ea6be07fa54cf9bdee3f18415805f17c1848444b33f8c99c1a8562R121-R154

Reply:
Or, the base class should not implement it, but that implementation should be in the visibility backend. Or is that what you meant?

Reply:
It's important to draw a clear distinction between the generic, core part and the actual rule implementation part.

Reply:
And this already has that excellent distinction! It's just a matter of compartmentalizing it properly.

Reply:
Indeed that is what I meant. I missed a few steps while refactoring out the "visibility" stuff to the backend so the core merely is the "dependency rules" part of it.

Reply:
Cool, ping me when you want me to take another look?

Reply:
@benjyw heh thanks!, just pushed the refactor now :)